src/__experimental__/components/date/date-picker.component.js

```diff
@@ -44,15 +45,22 @@ const DatePicker = (props) => {
   };
 
   useEffect(() => {
-    if (props.selectedDate && monthOrYearHasChanged(datepicker, props.selectedDate)) {
-      datepicker.current.showMonth(props.selectedDate);
+    if (hasComponentUpdated()) {
+      const updatedDate = isoFormattedValueString(props.inputDate);
+      datepicker.current.showMonth(DateHelper.stringToDate(updatedDate));
+      setCurrentInputDate(updatedDate);
     }
-  }, [props.selectedDate]);
+  }, [props.selectedDate, props.inputDate]);
 
   function handleDayClick(selectedDate, modifiers) {
     if (!modifiers.disabled) props.handleDateSelect(selectedDate);
   }
 
+  function hasComponentUpdated() {
+    const propDate = isoFormattedValueString(props.inputDate);
```

Comment:
what if props.inputDate is undefined?

Reply:
It shouldn't ever be undefined as the value passed from the input is based of the value in state, I just added this guard so that the picker showMonth is never going to be passed undefined

Reply:
Oh wait I realise what you mean now, I will move the guard 👍

Reply:
then why do we have a check for that in the line below?

Reply:
no need as it will just be undefined and falsy value will be returned by the guard below